### DISCUSSION TITLE
Datasource for Atlassian plugins

### DISCUSSION THREAD

**Discussion by pathob** [renovatebot/renovate#26260] [Suggest an Idea]
### Tell us more.

Hi,

short question: Would you accept a PR for an Atlassian plugin datasource?

Atlassian plugins (I'm referring to data center, not cloud) are distributed via the [Atlassian Marketplace](https://marketplace.atlassian.com). The Atlassian Marketplace has a REST API and [there is an endpoint to download versions](https://developer.atlassian.com/platform/marketplace/rest/v2/api-group-app-versions/#api-addons-addonkey-versions-get). The REST API requires to pass an `addonKey` that would serve as the package name. For example the versions of an app such as the [ScriptRunner for Jira](https://marketplace.atlassian.com/apps/6820/scriptrunner-for-jira?tab=overview&hosting=datacenter), whose `addonKey` is `com.onresolve.jira.groovy.groovyrunner`, can be downloaded using the following curl command:

```bash
curl --request GET \
  --url 'https://marketplace.atlassian.com/rest/2/addons/com.onresolve.jira.groovy.groovyrunner/versions' \
  --header 'Accept: application/json'
```

I guess that's only a few hours of work for an experienced Typescript and Renovate developer, but we would also try to do the contribution ourselves.

Thanks for your reply.

**Comment by viceice** [comment]
sure, go ahead

**Comment by secustor** [comment]
You can also look into [customDatasources](https://docs.renovatebot.com/modules/datasource/custom/).
